tuplex/python/tuplex/dataset.py

```diff
@@ -142,6 +172,26 @@ def show(self, nrows=None):
 
         self._dataSet.show(nrows)
 
+    def showHTMLPreview(self, topLimit=5, bottomLimit=5):
+        """ action that generates a physical plan, processes data and return a subset of results as nicely formatted
+        HTML table to stdout.
+
+        Args:
+            topLimit (int): number of top rows to collect. If ``None`` all rows will be collected
+            bottomLimit (int): number of bottom rows to collect. If ``None`` all rows will be collected
+
+        Returns:
+            string: an HTML table showing a preview of the data
+        """
+        assert self._dataSet is not None, 'internal API error, datasets must be created via context objects'
+
+        if topLimit is None or topLimit < 0:
+            topLimit = -1
+        if bottomLimit is None or bottomLimit < 0:
+            bottomLimit = -1
+
+        return self._dataSet.showHTMLPreview(topLimit, bottomLimit)
```

Comment:
I feel like you want to have the logic to turn everything into HTML implemented in python here (so it's easier to modify) and ship some efficient intermediate format from C++ to Python